Algorithm/QCAlgorithm.Trading.cs

```diff
@@ -36,18 +36,10 @@ public partial class QCAlgorithm
         /// <summary>
         /// Wait semaphore to signal the algoritm is currently processing a synchronous order.
         /// </summary>
-        public bool ProcessingOrder
+        public bool ProcessingEvents
```

Comment:
Indeed this or `volatile` is a good way to do this, however, I recently pushed https://github.com/QuantConnect/Lean/commit/9c1b0f4616e599f77e33f30c3c1ee8f6b4ef3c74 which removes this entirely. Instead there's a `ManualResetEventSlim` that can be used to wait for things to finish.